WordPress/Classes/ViewRelated/Blog/BlogSelectorViewController.m

```diff
@@ -186,6 +186,15 @@ - (IBAction)cancelButtonTapped:(id)sender
     }
     
     if (self.dismissOnCancellation) {
+        [self dismissViewController];
+    }
+}
+
+- (void)dismissViewController
+{
+    if (self.isModal && self.isRootInNavigation) {
+        [self dismissViewControllerAnimated:YES completion:nil];
+    } else {
```

Comment:
Besides my more generic comment above, from what I see this is only ever presented modally. Do we really need all these checks?